### PR TITLE
Basic functionality

### PR DESCRIPTION
**Pull request by aalekseevx**

### CODE REVIEW

File: app/automation.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    new_dfa.states = [State(
+        name=str(id_),
+        is_terminal=will_be_terminal[id_]
+    ) for id_ in range(next_component)]
+    new_dfa.start = str(component[dfa.start])
+    new_dfa.alphabet = dfa.alphabet
+
+    new_dfa.transmissions = []
+    for edge in dfa.transmissions:
+        c1 = component[edge.source]
```
Comment (akhtyamovpavel) [inline app/automation.py]: Maybe `source_component`?

**Review by akhtyamovpavel — APPROVED** [review]
Code is ok!

I consider that you are using NFA->DFA->min DFA conversion for your work?

**Comment by aalekseevx** [comment]
> Code is ok!
> 
> I consider that you are using NFA->DFA->min DFA conversion for your work?

Yes, that's the way it works.

File: app/automation.py (hunk 1 above)
Comment (aalekseevx) [reply app/automation.py]: Good idea, fixed that.

**Pull request merged**